### PR TITLE
feat(paymentMethod): Mutation deletePaymentMethod and insert paymentM…

### LINKED ISSUE #768: [API] Mutation removePaymentMethod

### PR DESCRIPTION
**Pull request by LittleWhiteYA**
…ethod in CheckoutSubscriptionWithPrime

Close #768 

## 這個 PR 是？ 

- 補上 mutation deletePaymentMethod
- 在 call mutation CheckoutSubscriptionWithPrime 的時候，同時插入 paymentMethod

## 我應該從何看起？  

https://hackmd.io/khxI9FJCSW-1QG8mUo7fTA

## 若可以手動測試，要如何測試？ 

先手動 insert paymentMethod 後再跑下面的 query

```gql
mutation {
  removePaymentMethod(id: "6384d9e77a1a925794ab894b")
}
```

### CODE REVIEW

**Comment by mark86092** [comment]
There are merge conflict, need merge latest dev

**Comment by mark86092** [comment]
Will we have unit test ?

File: src/schema/subscription_plan.js [hunk 1]
```diff
@@ -230,6 +231,26 @@ const resolvers = {
...
+                        "type",
+                        "issuer",
+                        "issuer_zh_tw",
+                        "expiry_date",
+                        "bin_code",
+                        "last_four",
+                    ]),
+                };
+
+                await PaymentMethod.create(insertPaymentMethod);
```
Comment (mark86092) [inline src/schema/subscription_plan.js:252]: `insertPaymentRecord.paymentMethodSnapshot` need to update too

Comment (LittleWhiteYA) [reply src/schema/subscription_plan.js:231]: 已改

File: src/schema/subscription_plan.js [hunk 2]
```diff
@@ -210,6 +226,10 @@ const resolvers = {
 
                 await Subscription.create(insertSubscription);
 
+                const { card_secret } = data;
```
Comment (mark86092) [inline src/schema/subscription_plan.js:229]: Should this be in L170?

Comment (LittleWhiteYA) [reply src/schema/subscription_plan.js:229]: yes, fixed

File: src/schema/paymentMethod.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+const Mutation = gql`
+    extend type Mutation {
+        removePaymentMethod(id: ID!): Boolean!
+    }
+`;
+
+const resolvers = {
+    Mutation: {
+        removePaymentMethod: combineResolvers(
+            isAuthenticated,
```
[the remark this answers is not in the document]
Comment (LittleWhiteYA) [reply src/schema/paymentMethod.js:21]: yes, addressed

**Pull request merged**